api-client/src/i18n/DateFormatter.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import type Moment from 'moment'
+import moment from 'moment'
+
+export type FormatFunctionType = (date: Moment, options: {| format?: string |}) => string
+
+class DateFormatter {
+  _fallbackFormat: string | void
+  _defaultLocale: string
+
+  constructor (fallbackFormat: string | void, defaultLocale: string) {
```

Comment:
Maybe it makes sense to change the order for easier construction:
```suggestion
  constructor (defaultLocale: string, fallbackFormat: string | void, ) {
```
That way we could use it this way:
```
new DateFormatter('de')
```